src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/reference/Reference.vue

```diff
@@ -91,8 +93,27 @@ export default defineComponent({
       }
     },
     handleSubmit() {
-      this.alertStore.setWarningAlert("Submit does not work yet");
+
+      this.alertStore.setWarningAlert("User Accepted");
     },
+    async handleDecline() {
+
+      const currentStepFormId = this.wizardStore.currentStep.form.id;
+      const formRef = (this.$refs.wizard as typeof Wizard).$refs[currentStepFormId][0].$refs[currentStepFormId];
+      const { valid } = await formRef.validate();
+
+      if (!valid) {
+        this.alertStore.setFailureAlert("Please fill out all required fields");
+        return;
+      }
+
+      const reason = this.wizardStore.wizardData[this.wizardStore.wizardConfig.steps.decline.form.inputs.referenceDecline.id];
+      const result = await optOutReference(this.$route.params.token as string, reason as Components.Schemas.UnabletoProvideReferenceReasons);
+      console.log(result);
```

Comment:
`console.log(result)` should be removed after debugging

Reply:
resolved